src/migrator/async.ts

```diff
@@ -64,19 +63,10 @@ const $run = async (
 	}> = [];
 
 	// get a transaction for setting up the async migrator
-	const modelIsNew = await sbvrUtils.isModelNew(setupTx, modelName);
-	const executedMigrations = await getExecutedMigrations(setupTx, modelName);
 
-	if (modelIsNew) {
-		(sbvrUtils.api.migrations?.logger.info ?? console.info)(
-			'First time model has executed, skipping async migrations',
-		);
+	const executedMigrations = await getExecutedMigrations(setupTx, modelName);
 
-		return await setExecutedMigrations(setupTx, modelName, [
-			...executedMigrations,
-			...Object.keys(migrations),
-		]);
-	}
+	// if the model is new, the sync migration parts (marked by finalize=true) are already marked in the sync migration runner.
```

Comment:
And we know that it'll definitely mark them as executed before we reach this branch?

Reply:
Yes definitively, the sync migrations are executed guaranteed before the async migrations. Any async migration that is finalized is executed in the sync migration block and will be marked as executed there.